src/command_modules/azure-cli-role/azure/cli/command_modules/role/custom.py

```diff
@@ -33,8 +34,12 @@
 
 _CUSTOM_RULE = 'CustomRole'
 
-def _auth_client_factory(**_):
-    return get_mgmt_service_client(AuthorizationManagementClient)
+def _auth_client_factory(resource_id=None):
+    subscription_id = None
+    matched = re.match('/subscriptions/(?P<subscription>[^/]*)/', resource_id)
+    if matched:
+        subscription_id = matched.groupdict()['subscription']
```

Comment:
Actually, the parse_resource_id doesn't work here. The resource_id could be a group id which the parsing logic can't handle.